addons/purchase_stock/static/tests/tours/purchase_catalog_suggest.js

```diff
@@ -0,0 +1,147 @@
+import { registry } from "@web/core/registry";
+import {
+    freezeDateTime,
+    selectPOVendor,
+    selectPOWarehouse,
+    goToCatalogFromPO,
+    goToPOFromCatalog,
+    toggleSuggest,
+    setSuggestParameters,
+} from "./tour_helpers";
+
+registry.category("web_tour.tours").add("test_purchase_catalog_suggest_search", {
+    steps: () => [
+        ...freezeDateTime("2021-01-14 09:12:15"), // Same date as python @freeze decorator
+        { trigger: ".o_purchase_order" },
+        {
+            content: "Create a New PO",
+            trigger: ".o_list_button_add",
+            run: "click",
+        },
+        ...selectPOVendor("Julia Agrolait"),
+        ...goToCatalogFromPO(),
+        toggleSuggest(),
+        { trigger: "button[name='toggle_suggest_catalog'].fa-toggle-off" },
+        {
+            content: "Check suggest fields hidden when suggest is off",
+            trigger: ".o_kanban_view.o_purchase_product_kanban_catalog_view",
+            run() {
+                const selectors = [
+                    ".o_TimePeriodSelectionField",
+                    "input.o_PurchaseSuggestInput",
+                    ".o_purchase_suggest_footer",
+                ];
+                const stillVisible = selectors.some((sel) => {
+                    const el = document.querySelector(sel);
+                    return el && el.offsetParent !== null;
+                });
+                if (stillVisible) {
+                    throw new Error("Toggle did not hide elements");
+                }
+            },
+        },
+        ...goToPOFromCatalog(),
+        ...goToCatalogFromPO(),
+        { trigger: "button[name='toggle_suggest_catalog'].fa-toggle-off" }, // Should still be off
+        toggleSuggest(),
+        { trigger: "button[name='toggle_suggest_catalog'].fa-toggle-on" },
+        ...setSuggestParameters({ basedOn: "Last 3 months", nbDays: 90, factor: 100, wait: 1000 }),
+        {
+            trigger: "span[name='suggest_total']",
+            async run() {
+                await new Promise((r) => setTimeout(r, 1000));
+                const total = parseFloat(this.anchor.textContent);
+                if (total !== 0) {
+                    throw new Error(`Expected suggest_total = 0, got ${total} (wrong warehouse)`);
+                }
+            },
+        },
+        ...goToPOFromCatalog(),
+        ...selectPOWarehouse("Base Warehouse: Receipts"), // Still the same PO, no need to reset vendor
+        ...goToCatalogFromPO(),
+        { trigger: ".o_kanban_view.o_purchase_product_kanban_catalog_view" },
+        { trigger: "button[name='toggle_suggest_catalog'].fa-toggle-on" }, // Should still be ON
+        ...setSuggestParameters({ basedOn: "Last 7 days", nbDays: 28, factor: 50, wait: 1000 }),
+        {
+            trigger: "span[name='suggest_total']",
+            async run() {
+                await new Promise((r) => setTimeout(r, 1000));
```

Comment:
Too much wait, you can check the suggest total has the right amount directly in the trigger (check `:contains`)